Virtual Scroll on log tab

ToxicKevinFerm opened the pull request:
Don't know if we want to do something with this @kayla-glick but rendering x amount of thousand elementsfor the log result isn't very performant.
This is not perfect, but we keep only 50 elements at a time like this.

kayla-glick commented:
Virtual scroll would be great! But it looks like there's maybe some issues to iron out?

It looks like a lot of the styling is missing

Old
<img width="1348" alt="image" src="https://github.com/wowsims/cata/assets/12898988/29050c83-278b-4ab4-a109-1053d50b17ce">

New
<img width="1230" alt="image" src="https://github.com/wowsims/cata/assets/12898988/5ee16c7d-fb2c-43de-9eaf-a318409d94d1">

And I'm not sure if this is just an issue with sims not being implemented, but the log is cutting off very early
<img width="1065" alt="image" src="https://github.com/wowsims/cata/assets/12898988/593074b2-33a6-4373-83fe-a8370be39489">

ToxicKevinFerm commented:
> Virtual scroll would be great! But it looks like there's maybe some issues to iron out?

Definitely has some issues. Just seeing if it would be worth spending more time fixing them. But it seems so, so I'll look at it :D

ToxicKevinFerm commented:
![image](https://github.com/wowsims/cata/assets/95472394/bf2c099d-b335-48a4-9564-d43ae78c90be)

Fixed styling and added a basic filter, splits on space and takes quotation marks for more specific filtering. Also a scroll to top button.
